src/modules/cmdpal/Microsoft.CmdPal.UI/MainWindow.xaml.cs

```diff
@@ -1006,4 +1018,44 @@ public void Dispose()
         _localKeyboardListener.Dispose();
         DisposeAcrylic();
     }
+
+    public void Receive(DragStartedMessage message)
+    {
+        _preventHideWhenDeactivated = true;
+    }
+
+    public void Receive(DragCompletedMessage message)
+    {
+        _preventHideWhenDeactivated = false;
+        Task.Delay(200).ContinueWith(_ =>
+        {
+            DispatcherQueue.TryEnqueue(ForceActivate);
+        });
+    }
+
+    private unsafe void ForceActivate()
+    {
+        var foregroundWindow = PInvoke.GetForegroundWindow();
+        if (foregroundWindow == _hwnd)
+        {
+            return;
+        }
+
+        // This is bad, evil, and I'll have to forgo today's dinner dessert to punish myself
```

Comment:
y'know, I've written this code before, so it can't be _that_ evil 😝 
https://github.com/microsoft/terminal/blob/224ac9de4758f67b7f83b74909d1c60f15715a3c/src/cascadia/WindowsTerminal/IslandWindow.cpp#L1460-L1474